### DISCUSSION TITLE
关于FastSpeech2 with CSMSC训练

### DISCUSSION THREAD

**Discussion by syy-love** [PaddlePaddle/PaddleSpeech#2349] [Q&A] [answered]
跑到这一步时
![3](https://user-images.githubusercontent.com/51405724/188427195-232c1ff1-ec1f-4877-8860-10d2b7aa43d7.jpg)
总会报这个错误
![2](https://user-images.githubusercontent.com/51405724/188427251-6db4d26d-1952-4099-8991-60ff866bb5b3.jpg)
之前是能跑通的，有无大佬帮分析一下原因
paddle版本：paddlepaddle-gpu==2.3.1

**Comment by yt605155624** [comment, marked as answer]
我在 2022.09.05 拉取的 paddle develop 和 paddle2onnx 1.0.0 下，用预训练模型执行如下命令，无法复现你的问题
```bash
wget https://paddlespeech.bj.bcebos.com/Parakeet/released_models/fastspeech2/fastspeech2_cnndecoder_csmsc_ckpt_1.0.0.zip
unzip fastspeech2_cnndecoder_csmsc_ckpt_1.0.0.zip
mkdir -p exp/cnndecoder/checkpoints
cp fastspeech2_cnndecoder_csmsc_ckpt_1.0.0/snapshot_iter_153000.pdz exp/cnndecoder/checkpoints
mkdir -p dump/train
cp fastspeech2_cnndecoder_csmsc_ckpt_1.0.0/speech_stats.npy dump/train
cp fastspeech2_cnndecoder_csmsc_ckpt_1.0.0/phone_id_map.txt dump
# 修改 run_cnndecoder.sh 里的 ckpt_name
wget https://paddlespeech.bj.bcebos.com/Parakeet/released_models/pwgan/pwg_baker_ckpt_0.4.zip
unzip pwg_baker_ckpt_0.4.zip
./run_cnndecoder.sh --stage 3 --stop-stage 10
```
你在 86 行报错是 am warm up 时就有问题，可能是你自己训练的和我们的参数不匹配， 此处的输入数值是根据我们的预训练模型定义的，如 phone_id_map.txt 有 268 行所以此处选择 266，你需要按照自己的模型定义这里的数值，https://github.com/PaddlePaddle/PaddleSpeech/blob/ea9ee93739f023916ef7736ad8f4d8d3a34d4f8a/paddlespeech/t2s/exps/ort_predict_e2e.py#L77
可以用 https://netron.app/ 可视化你导出的 onnx 模型确定参数类型和范围
